### PR TITLE
Create user facade

### PR DESCRIPTION
**Pull request by vecerap**

### CODE REVIEW

**Comment by vecerap** [comment]
@BRTEK007 @KoubaFromPucks Do we need more methods in this facade?

**Comment by KoubaFromPucks** [comment]
> @BRTEK007 @KoubaFromPucks Do we need more methods in this facade?

Maybe `GetCompletedQuizzes?`

**Comment by brtkkndrl** [comment]
> @BRTEK007 @KoubaFromPucks Do we need more methods in this facade?

Can't think of any right now.

**Comment by brtkkndrl** [comment]
> > @BRTEK007 @KoubaFromPucks Do we need more methods in this facade?
> 
> Maybe `GetCompletedQuizzes?`

Shouldn't that be in QuizFacade?

**Comment by KoubaFromPucks** [comment]
> > > @BRTEK007 @KoubaFromPucks Do we need more methods in this facade?
> > 
> > 
> > Maybe `GetCompletedQuizzes?`
> 
> Shouldn't that be in QuizFacade?

Well it makes a sense

File: src/Quizzer.API.BL/Facades/UserFacade.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    public class UserFacade : FacadeBase<UserEntity, UserDetailModel> {
+        public UserFacade(IUnitOfWorkFactory unitOfWorkFactory, IMapper mapper) : base(unitOfWorkFactory, mapper) {
+        }
+
+        public override Guid Save(UserDetailModel model) {
+            if (string.IsNullOrEmpty(model.Name?.Trim())) {
+                throw new ArgumentException("Name cannot be empty");
+            }
+
+            return base.Save(model);
```
Comment (KoubaFromPucks) [inline src/Quizzer.API.BL/Facades/UserFacade.cs]: Saving untrimed name to database

Comment (vecerap) [reply src/Quizzer.API.BL/Facades/UserFacade.cs]: Fixed.

**Review by KoubaFromPucks — APPROVED** [review]

**Review by brtkkndrl — APPROVED** [review]

**Pull request merged**